src/OrbitBase/include/OrbitBase/JoinFutures.h

```diff
@@ -90,6 +149,18 @@ Future<std::vector<T>> JoinFutures(absl::Span<const Future<T>> futures) {
 
   return shared_state->GetFuture();
 }
+
+// Returns a future which completes when all given futures have completed.
+// The returning future will contain a tuple of all the given future's values.
+// Note that a future of type `Future<void>` is not supported as an argument.
+template <typename Arg0, typename... Args>
+[[nodiscard]] Future<std::tuple<Arg0, Args...>> JoinFutures(Future<Arg0> future0,
+                                                            Future<Args>... futures) {
+  return orbit_base_internal::JoinFuturesTupleImpl<Arg0, Args...>(
+      std::move(future0), std::move(futures)...,
+      std::make_index_sequence<1 + sizeof...(futures)>());
+}
```

Comment:
Out of curiosity: Why is this extra Arg0 future0 necessary?

Reply:
Because a parameter pack can also be deduced to zero arguments. If you want to have at least one argument then `Arg0` is needed.

Reply:
That makes sense! Thank you